ui/components/app/snaps/snap-ui-renderer/components/field.ts

```diff
@@ -16,12 +16,29 @@ export const field: UIComponentFactory<FieldElement> = ({ element, form }) => {
   const child = children[0] as JSXElement;
 
   switch (child.type) {
+    case 'FileInput': {
+      return {
+        element: 'SnapUIFileInput',
+        props: {
+          id: child.props.name,
```

Comment:
I think you are gonna need a cast here

Reply:
Not sure why that would be necessary?

Reply:
Hmm, I thought it failed to infer the type in the past but maybe not anymore